File: app/src/main/java/com/redteamobile/smart/recycler/ProfileAdapter.java
package com.redteamobile.smart.recycler;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.redteamobile.smart.R;
import com.redteamobile.smart.agent.AgentService;

import java.util.List;

public class ProfileAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private AgentService agentService;
    private final List<ProfileModel> profileList;

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_profile, parent, false);
        return new ProfileHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        if (holder instanceof ProfileHolder) {
            ProfileModel model = (ProfileModel) profileList.get(position);
            ((ProfileHolder) holder).bind(model);
        }
    }

    @Override
    public int getItemCount() {
        return profileList.size();
    }

    public ProfileAdapter(List<ProfileModel> profileList) {
        super();
        this.profileList = profileList;
    }

    public void setAgentService(AgentService agentService) {
        this.agentService = agentService;
    }

    public final class ProfileHolder extends RecyclerView.ViewHolder {

        public void bind(final ProfileModel profileModel) {
            View view = this.itemView;
            TextView tv_vitemIccid = (TextView) view.findViewById(R.id.itemIccid);
            tv_vitemIccid.setText((CharSequence) profileModel.getIccid());
            TextView tv_itemEnableBtn = (TextView) view.findViewById(R.id.itemEnableBtn);
            if (profileModel.getState() == 0) {
                tv_itemEnableBtn.setText(view.getResources().getString(R.string.enable_profile));
            } else {
                tv_itemEnableBtn.setText((CharSequence) view.getResources().getString(R.string.disable_profile));
            }

            tv_itemEnableBtn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (agentService != null) {
                        if (profileModel.getState() == 0) {
                            profileModel.setState(1);
                            agentService.enableProfile(profileModel.getIccid());

                        } else {
                            profileModel.setState(0);
                            agentService.disableProfile(profileModel.getIccid());

                        }
                    }
                }
            });
        }

        public ProfileHolder(View itemView) {
            super(itemView);
        }

    }
}

File: app/src/main/java/com/redteamobile/smart/agent/recycler/ProfileAdapter.kt
package com.redteamobile.smart.agent.recycler

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.redteamobile.smart.R
import com.redteamobile.smart.agent.AgentService
import kotlinx.android.synthetic.main.item_profile.view.*

class ProfileAdapter(private var profileList: List<ProfileModel>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private var agentService: AgentService? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return ProfileHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.item_profile, parent, false)
        )
    }

    override fun getItemCount(): Int = profileList.size

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is ProfileHolder) {
            val any = profileList[position]
            if (any is ProfileModel) {
                holder.bind(any)
            }
        }
    }

    fun setAgentService(agentService: AgentService) {
        this.agentService = agentService
    }

    inner class ProfileHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bind(profileModel: ProfileModel) {
            with(itemView) {
                itemIccid.text = profileModel.iccid
                if (profileModel.state == 0) {
                    itemEnableBtn.text = "启用"
                } else {
                    itemEnableBtn.text = "禁用"
                }
                if (profileModel.type == 0) {

                }
                itemEnableBtn.setOnClickListener {
                    if (agentService != null) {
                        if (profileModel.state == 0) {
                            profileModel.state = 1
                            agentService!!.enableProfile(profileModel.iccid)
                        } else {
                            profileModel.state = 0
                            agentService!!.disableProfile(profileModel.iccid)
                        }
                        notifyDataSetChanged()
                    }
                }
            }
        }
    }

}
